config/printer-anycubic-vyper-2021.cfg

```diff
@@ -141,3 +169,139 @@ gcode:
     G4 P300
     SET_PIN PIN=probe_reset_pin VALUE=1
     G4 P100
+
+[virtual_sdcard]
+path: ~/gcode_files
+
+[pause_resume]
+
+
+#Klipper required macros
+[gcode_macro PAUSE]
+description: Pause the actual running print
+rename_existing: PAUSE_BASE
+gcode:
+    {% set e = params.E|default(10) %}        #edit to your retract length
+    PAUSE_BASE
+    {% if printer.extruder.can_extrude|lower == 'true' %}
+      G1 E-{e} F2100
+    {% else %}
+      {action_respond_info("Extruder not hot enough")}
+    {% endif %}
+    END_POS
+    M140 S55
+
+[gcode_macro RESUME]
+description: Resume the actual running print
+rename_existing: RESUME_BASE
+gcode:
+    M140 S55
+    M104 S190
+    M190 S55
+    M109 S190
+    ##### set defaults #####
+    {% set e = params.E|default(5) %} #edit to your retract length
+    #### get VELOCITY parameter if specified ####
+    {% if 'VELOCITY' in params|upper %}
+      {% set get_params = ('VELOCITY=' + params.VELOCITY)  %}
+    {%else %}
+      {% set get_params = "" %}
+    {% endif %}
+    ##### end of definitions #####
+    G91
+    {% if printer.extruder.can_extrude|lower == 'true' %}
+      G1 E{e} F2100
+    {% else %}
+      {action_respond_info("Extruder not hot enough")}
+    {% endif %}
+    RESUME_BASE {get_params}
```

Comment:
Not required. Optional. Not for base config files

Reply:
same, this one is base on the current file. I just make it better to use. And I really think that is neccessary because when you pause the print, it just stop there. 

Those Gcode simulate how anycubic vyper move without klipper. In My view, Pause and Resume should be kept. But if not, maybe we need to delete the code in the current file.